src/ActionServer_FJT.c

```diff
@@ -494,19 +630,63 @@ void Ros_ActionServer_FJT_Goal_Complete(GOAL_END_TYPE goal_end_type)
         //-----------------------------------------------------------------------
         //check to see if each axis is in the desired location
         BOOL positionOk = TRUE;
-        int maxAxes = MAX_CONTROLLABLE_GROUPS * MP_GRP_AXES_NUM;
-        double violators[MAX_CONTROLLABLE_GROUPS * MP_GRP_AXES_NUM];
+        int numAxesToCheck = feedback_FollowJointTrajectory.feedback.joint_names.size;
 
+        //exploit GCC's VLA support. Not portable, but we're explicitly only
+        //supporting M+ GCC here with MotoROS2, so this should be acceptable
+        double violators[numAxesToCheck];
+        double posTolerance[numAxesToCheck];
         bzero(violators, sizeof(violators));
-        for (int axis = 0; axis < maxAxes; axis += 1)
+        bzero(posTolerance, sizeof(posTolerance));
+
+        //'parse' the JointTolerance elements from the goal. Map their 'name:tolerance'
+        //to the 'joint_index:tolerance' we need
+        STATUS status = Ros_ActionServer_FJT_Parse_GoalPosTolerances(
+            &g_actionServer_FJT_SendGoal_Request.goal.goal_tolerance,
+            &feedback_FollowJointTrajectory.feedback.joint_names,
+            posTolerance, numAxesToCheck);
+
+        if (status != OK)
         {
-            diff = fabs(feedback_FollowJointTrajectory.feedback.desired.positions.data[axis] - feedback_FollowJointTrajectory.feedback.actual.positions.data[axis]);
+            Ros_Debug_BroadcastMsg("%s: parsing 'goal_tolerance' field failed: %d", __func__, status);
+            //TODO(gavanderhoorn): what do we do here?
```

Comment:
what should we do here? @yai-rosejo and @ted-miller any ideas?

Reply:
IMO, `positionOk = FALSE;`
A trajectory should not 'pass' a tolerance-check if the tolerance cannot be properly evaluated.

Reply:
For this line and also this one:
https://github.com/Yaskawa-Global/motoros2/blob/64c652409b88eab479940a9d61019ef99dfa6458/src/ActionServer_FJT.c#L662-L673
Could we create a bool called parsingOk that way the checks down here:
https://github.com/Yaskawa-Global/motoros2/blob/64c652409b88eab479940a9d61019ef99dfa6458/src/ActionServer_FJT.c#L720-L769
Could have a branch for parsingOk similar to the ones for positionOk

Reply:
So add a new entry into the `Failed_Trajectory_Status` enum and give a unique error message indicating that the tolerance input was fubar?

Logically, that makes sense. But as mentioned above, the validity of the tolerances should be checked in `Ros_ActionServer_FJT_Goal_Received`. At this point, in `Ros_ActionServer_FJT_Goal_Received`, I think we should just fail with `Final position was outside tolerance`

Reply:
Ok, 'for now', I've implemented something in c552d58b248a3559c5cc5280ced8db22cdac32b1.

I don't like it too much, but it does result in a clear(er) error message than just "tolerance violation", and resolves the issue.